src/natcap/invest/urban_nature_access.py

```diff
@@ -254,103 +282,133 @@ def execute(args):
         target_path_list=[file_registry['aligned_population']],
         task_name='Resample population to LULC resolution')
 
-    greenspace_lulc_lookup = utils.build_lookup_from_csv(
-        args['lulc_attribute_table'], 'lucode')
-    squared_pixel_area = abs(numpy.multiply(*squared_lulc_pixel_size))
-    greenspace_area_map = {
-        lucode: int(attributes['greenspace']) * squared_pixel_area
-        for lucode, attributes in greenspace_lulc_lookup.items()
-    }
-    greenspace_area_map[lulc_raster_info['nodata'][0]] = FLOAT32_NODATA
-    greenspace_reclassification_task = graph.add_task(
-        utils.reclassify_raster,
-        kwargs={
-            'raster_path_band': (file_registry['aligned_lulc'], 1),
-            'value_map': greenspace_area_map,
-            'target_raster_path': file_registry['greenspace_area'],
-            'target_datatype': gdal.GDT_Float32,
-            'target_nodata': FLOAT32_NODATA,
-            'error_details': {
-                'raster_name': ARGS_SPEC['args']['lulc_raster_path']['name'],
-                'column_name': 'greenspace',
-                'table_name': ARGS_SPEC['args'][
-                    'lulc_attribute_table']['name'],
-            },
-        },
-        target_path_list=[file_registry['greenspace_area']],
-        task_name='Identify the area of greenspace in pixels',
-        dependent_task_list=[lulc_alignment_task]
-    )
-
-    search_radius_in_pixels = abs(
-        float(args['search_radius']) / squared_lulc_pixel_size[0])
-    kernel_creation_task = graph.add_task(
-        # All kernel creation types have the same function signature
-        kernel_types[kernel_type],
-        args=(search_radius_in_pixels, file_registry['kernel']),
-        kwargs={'normalize': False},  # Model math calls for un-normalized
-        task_name=f'2SFCA - Create {kernel_type} kernel',
-        target_path_list=[file_registry['kernel']]
-    )
-
-    convolved_population_task = graph.add_task(
-        _convolve_and_set_lower_bounds_for_population,
+    preprocess_attr_table_task = graph.add_task(
+        _preprocess_lulc_attribute_table,
         kwargs={
-            'signal_path_band': (file_registry['aligned_population'], 1),
-            'kernel_path_band': (file_registry['kernel'], 1),
-            'target_path': file_registry['convolved_population'],
-            'working_dir': intermediate_dir,
+            'source_attr_table_path': args['lulc_attribute_table'],
```

Comment:
I like how you're using all the args as `kwargs` even though they don't have to be. It's nice to see the arg variable name alongside its value.

Reply:
Thanks!  I've found it really helpful and clear when writing this out and it helps cut down on issues where I accidentally pass in the wrong thing.

The next nice-to-have would be to have my editor populate the kwargs ... not there yet.